xray/conventions.py

```diff
@@ -136,12 +136,16 @@ def decode_cf_datetime(num_dates, units, calendar=None):
             raise OutOfBoundsDatetime
 
         delta = _netcdf_to_numpy_timeunit(delta)
-        ref_date = pd.Timestamp(ref_date)
+        try:
+            ref_date = pd.Timestamp(ref_date)
+        except ValueError:
+            # ValueError is raised by pd.Timestamp for non-ISO timestamp
+            # strings, in which case we fall back to using netCDF4
+            raise OutOfBoundsDatetime
 
         dates = (pd.to_timedelta(flat_num_dates, delta) + ref_date).values
-    # ValueError is raised by pd.Timestamp for non-ISO timestamp strings,
-    # in which case we fall back to using netCDF4
-    except (OutOfBoundsDatetime, ValueError, OverflowError):
+
+    except (OutOfBoundsDatetime, OverflowError):
```

Comment:
This will give us a more interpretable error on old versions of pandas (e.g., as we get on read the docs). In that case, `ValueError` was being raised by `to_timedelta` instead of `Timestamp`. Basically this is just better form -- we're catching the error in exactly the place where it can go wrong. We already have test coverage for that.